src/values.cpp

```diff
@@ -792,6 +790,115 @@ Value::operator bool() const {
     return std::visit([](const auto& value) { return bool(value); }, this->impl->val);
 }
 
+static std::string pattern1 = R"((\s*-?\d+\.?\d*))";
+static std::string pattern2 = R"((\s*-?0[xX][\dA-Fa-f]+\.?[\dA-Fa-f]*))";
+static std::string pattern3 = R"((\s*-?\d+\.?\d*[eE]-?\d+))";
```

Comment:
maybe rename patterns to match the number-type so that you can easily recognise what type of number the pattern is used for.